portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/shoutbox")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    PrintWriter out = response.getWriter();
+    out.println("<h1>Log In</h1>");
+
+    // Only logged-in users can see the form
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      out.println("<p>Hello " + userService.getCurrentUser().getEmail() + "!</p>");
```

Comment:
You should try to avoid manually building HTML with strings as much as possible. It tends to be quite fragile and difficult to make sure it's correct over time. 

Some better options are to specify a template of the HTML in a file or even a single variable. Then you just fill the values later when you're ready to actually use it. Check out Angela's repo; I got her to follow this approach using java's Resource API.